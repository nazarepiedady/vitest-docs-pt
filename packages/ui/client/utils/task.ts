import type { Suite, Task } from '#types'

export function isSuite(task: Task): task is Suite {
  return Object.hasOwnProperty.call(task, 'tasks')
}

export function caseInsensitiveMatch(target: string, str2: string) {
  if (typeof target !== 'string' || typeof str2 !== 'string')
    return false
  return target.toLowerCase().includes(str2.toLowerCase())
}
