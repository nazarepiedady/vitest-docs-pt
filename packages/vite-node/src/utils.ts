import { fileURLToPath, pathToFileURL } from 'node:url'
import { existsSync } from 'node:fs'
import { resolve } from 'pathe'
import type { Arrayable, Nullable } from './types'

export const isWindows = process.platform === 'win32'

export function slash(str: string) {
  return str.replace(/\\/g, '/')
}

export const VALID_ID_PREFIX = '/@id/'

export function normalizeRequestId(id: string, base?: string): string {
  if (base && id.startsWith(base))
    id = `/${id.slice(base.length)}`

  return id
    .replace(/^\/@id\/__x00__/, '\0') // virtual modules start with `\0`
    .replace(/^\/@id\//, '')
    .replace(/^__vite-browser-external:/, '')
    .replace(/^file:/, '')
    .replace(/^\/+/, '/') // remove duplicate leading slashes
    .replace(/\?v=\w+/, '?') // remove ?v= query
    .replace(/&v=\w+/, '') // remove &v= query
    .replace(/\?t=\w+/, '?') // remove ?t= query
    .replace(/&t=\w+/, '') // remove &t= query
    .replace(/\?import/, '?') // remove ?import query
    .replace(/&import/, '') // remove &import query
    .replace(/\?&/, '?') // replace ?& with just ?
    .replace(/\?+$/, '') // remove end query mark
}

export const queryRE = /\?.*$/s
export const hashRE = /#.*$/s

export const cleanUrl = (url: string): string =>
  url.replace(hashRE, '').replace(queryRE, '')

const internalRequests = [
  '@vite/client',
  '@vite/env',
]

const internalRequestRegexp = new RegExp(`^/?(${internalRequests.join('|')})$`)

export const isInternalRequest = (id: string): boolean => {
  return internalRequestRegexp.test(id)
}

export function normalizeModuleId(id: string) {
  return id
    .replace(/\\/g, '/')
    .replace(/^\/@fs\//, isWindows ? '' : '/')
    .replace(/^file:\//, '/')
    .replace(/^node:/, '')
    .replace(/^\/+/, '/')
}

export function isPrimitive(v: any) {
  return v !== Object(v)
}

export function toFilePath(id: string, root: string): string {
  let absolute = (() => {
    if (id.startsWith('/@fs/'))
      return id.slice(4)
    if (!id.startsWith(root) && id.startsWith('/')) {
      const resolved = resolve(root, id.slice(1))
      // The resolved path can have query values. Remove them before checking
      // the file path.
      if (existsSync(resolved.replace(/\?.*$/, '')))
        return resolved
    }
    return id
  })()

  if (absolute.startsWith('//'))
    absolute = absolute.slice(1)

  // disambiguate the `<UNIT>:/` on windows: see nodejs/node#31710
  return isWindows && absolute.startsWith('/')
    ? slash(fileURLToPath(pathToFileURL(absolute.slice(1)).href))
    : absolute
}

/**
 * Convert `Arrayable<T>` to `Array<T>`
 *
 * @category Array
 */
export function toArray<T>(array?: Nullable<Arrayable<T>>): Array<T> {
  if (array === null || array === undefined)
    array = []

  if (Array.isArray(array))
    return array

  return [array]
}
