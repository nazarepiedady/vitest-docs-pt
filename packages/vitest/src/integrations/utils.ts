// Runtime utils exposed to `vitest`

export function getRunningMode() {
  return process.env.VITEST_MODE === 'WATCH' ? 'watch' : 'run'
}

export function isWatchMode() {
  return getRunningMode() === 'watch'
}
