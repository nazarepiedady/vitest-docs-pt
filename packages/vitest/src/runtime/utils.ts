import type { DoneCallback } from '../types'

/**
 * A simple wrapper for converting callback style to promise
 */
export function withCallback(fn: (done: DoneCallback) => void): Promise<void> {
  return new Promise((resolve, reject) =>
    fn((err) => {
      if (err)
        reject(err)
      else
        resolve()
    }),
  )
}
