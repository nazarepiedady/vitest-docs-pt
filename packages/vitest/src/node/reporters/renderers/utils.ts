import { basename, dirname, isAbsolute, relative } from 'pathe'
import c from 'picocolors'
import stripAnsi from 'strip-ansi'
import type { SnapshotSummary, Task } from '../../../types'
import { slash } from '../../../utils'
import { F_CHECK, F_CROSS, F_DOT, F_DOWN, F_DOWN_RIGHT, F_LONG_DASH, F_POINTER } from '../../../utils/figures'
import type { SuiteHooks } from './../../../types/tasks'

export const spinnerMap = new WeakMap<Task, () => string>()
export const hookSpinnerMap = new WeakMap<Task, Map<string, () => string>>()
export const pointer = c.yellow(F_POINTER)
export const skipped = c.dim(c.gray(F_DOWN))

export function getCols(delta = 0) {
  let length = process.stdout?.columns
  if (!length || isNaN(length))
    length = 30
  return Math.max(length + delta, 0)
}

export function divider(text?: string, left?: number, right?: number) {
  const cols = getCols()

  if (text) {
    const textLength = stripAnsi(text).length
    if (left == null && right != null) {
      left = cols - textLength - right
    }
    else {
      left = left ?? Math.floor((cols - textLength) / 2)
      right = cols - textLength - left
    }
    left = Math.max(0, left)
    right = Math.max(0, right)
    return `${F_LONG_DASH.repeat(left)}${text}${F_LONG_DASH.repeat(right)}`
  }
  return F_LONG_DASH.repeat(cols)
}

export function formatTestPath(root: string, path: string) {
  if (isAbsolute(path))
    path = relative(root, path)

  const dir = dirname(path)
  const ext = path.match(/(\.(spec|test)\.[cm]?[tj]sx?)$/)?.[0] || ''
  const base = basename(path, ext)

  return slash(c.dim(`${dir}/`) + c.bold(base)) + c.dim(ext)
}

export function renderSnapshotSummary(rootDir: string, snapshots: SnapshotSummary) {
  const summary: string[] = []

  if (snapshots.added)
    summary.push(c.bold(c.green(`${snapshots.added} written`)))
  if (snapshots.unmatched)
    summary.push(c.bold(c.red(`${snapshots.unmatched} failed`)))
  if (snapshots.updated)
    summary.push(c.bold(c.green(`${snapshots.updated} updated `)))

  if (snapshots.filesRemoved) {
    if (snapshots.didUpdate)
      summary.push(c.bold(c.green(`${snapshots.filesRemoved} files removed `)))

    else
      summary.push(c.bold(c.yellow(`${snapshots.filesRemoved} files obsolete `)))
  }

  if (snapshots.filesRemovedList && snapshots.filesRemovedList.length) {
    const [head, ...tail] = snapshots.filesRemovedList
    summary.push(`${c.gray(F_DOWN_RIGHT)} ${formatTestPath(rootDir, head)}`)

    tail.forEach((key) => {
      summary.push(`  ${c.gray(F_DOT)} ${formatTestPath(rootDir, key)}`)
    })
  }

  if (snapshots.unchecked) {
    if (snapshots.didUpdate)
      summary.push(c.bold(c.green(`${snapshots.unchecked} removed`)))

    else
      summary.push(c.bold(c.yellow(`${snapshots.unchecked} obsolete`)))

    snapshots.uncheckedKeysByFile.forEach((uncheckedFile) => {
      summary.push(`${c.gray(F_DOWN_RIGHT)} ${formatTestPath(rootDir, uncheckedFile.filePath)}`)
      uncheckedFile.keys.forEach(key => summary.push(`  ${c.gray(F_DOT)} ${key}`))
    })
  }

  return summary
}

export function countTestErrors(tasks: Task[]) {
  return tasks.reduce((c, i) => c + (i.result?.errors?.length || 0), 0)
}

export function getStateString(tasks: Task[], name = 'tests', showTotal = true) {
  if (tasks.length === 0)
    return c.dim(`no ${name}`)

  const passed = tasks.filter(i => i.result?.state === 'pass')
  const failed = tasks.filter(i => i.result?.state === 'fail')
  const skipped = tasks.filter(i => i.mode === 'skip')
  const todo = tasks.filter(i => i.mode === 'todo')

  return [
    failed.length ? c.bold(c.red(`${failed.length} failed`)) : null,
    passed.length ? c.bold(c.green(`${passed.length} passed`)) : null,
    skipped.length ? c.yellow(`${skipped.length} skipped`) : null,
    todo.length ? c.gray(`${todo.length} todo`) : null,
  ].filter(Boolean).join(c.dim(' | ')) + (showTotal ? c.gray(` (${tasks.length})`) : '')
}

export function getStateSymbol(task: Task) {
  if (task.mode === 'skip' || task.mode === 'todo')
    return skipped

  if (!task.result)
    return c.gray('·')

  // pending
  if (task.result.state === 'run') {
    if (task.type === 'suite')
      return pointer
    let spinner = spinnerMap.get(task)
    if (!spinner) {
      spinner = elegantSpinner()
      spinnerMap.set(task, spinner)
    }
    return c.yellow(spinner())
  }

  if (task.result.state === 'pass') {
    return task.type === 'benchmark'
      ? c.green(F_DOT)
      : c.green(F_CHECK)
  }

  if (task.result.state === 'fail') {
    return task.type === 'suite'
      ? pointer
      : c.red(F_CROSS)
  }

  return ' '
}

export function getHookStateSymbol(task: Task, hookName: keyof SuiteHooks) {
  const state = task.result?.hooks?.[hookName]

  // pending
  if (state && state === 'run') {
    let spinnerMap = hookSpinnerMap.get(task)
    if (!spinnerMap) {
      spinnerMap = new Map<string, () => string>()
      hookSpinnerMap.set(task, spinnerMap)
    }
    let spinner = spinnerMap.get(hookName)
    if (!spinner) {
      spinner = elegantSpinner()
      spinnerMap.set(hookName, spinner)
    }
    return c.yellow(spinner())
  }
}

export const spinnerFrames = process.platform === 'win32'
  ? ['-', '\\', '|', '/']
  : ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏']

export function elegantSpinner() {
  let index = 0

  return () => {
    index = ++index % spinnerFrames.length
    return spinnerFrames[index]
  }
}

export function duration(time: number, locale = 'en-us') {
  if (time < 1e0)
    return `${Number((time * 1e3).toFixed(2)).toLocaleString(locale)} ps`

  if (time < 1e3)
    return `${Number(time.toFixed(2)).toLocaleString(locale)} ns`
  if (time < 1e6)
    return `${Number((time / 1e3).toFixed(2)).toLocaleString(locale)} µs`
  if (time < 1e9)
    return `${Number((time / 1e6).toFixed(2)).toLocaleString(locale)} ms`
  if (time < 1e12)
    return `${Number((time / 1e9).toFixed(2)).toLocaleString(locale)} s`
  if (time < 36e11)
    return `${Number((time / 60e9).toFixed(2)).toLocaleString(locale)} m`

  return `${Number((time / 36e11).toFixed(2)).toLocaleString(locale)} h`
}

export function formatTimeString(date: Date) {
  return date.toTimeString().split(' ')[0]
}

export function formatProjectName(name: string | undefined, suffix = ' ') {
  if (!name)
    return ''
  const index = name.split('').reduce((acc, v, idx) => acc + v.charCodeAt(0) + idx, 0)
  const colors = [
    c.blue,
    c.yellow,
    c.cyan,
    c.green,
    c.magenta,
  ]
  return colors[index % colors.length](`|${name}|`) + suffix
}
