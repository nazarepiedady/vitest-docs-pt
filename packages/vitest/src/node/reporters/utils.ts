import type { ViteNodeRunner } from 'vite-node/client'
import type { Reporter } from '../../types'
import { ensurePackageInstalled } from '../../utils'
import { BenchmarkReportsMap, ReportersMap } from './index'
import type { BenchmarkBuiltinReporters, BuiltinReporters } from './index'

async function loadCustomReporterModule<C extends Reporter>(path: string, runner: ViteNodeRunner): Promise<new () => C> {
  let customReporterModule: { default: new () => C }
  try {
    customReporterModule = await runner.executeId(path)
  }
  catch (customReporterModuleError) {
    throw new Error(`Failed to load custom Reporter from ${path}`, { cause: customReporterModuleError as Error })
  }

  if (customReporterModule.default === null || customReporterModule.default === undefined)
    throw new Error(`Custom reporter loaded from ${path} was not the default export`)

  return customReporterModule.default
}

function createReporters(reporterReferences: Array<string | Reporter | BuiltinReporters>, runner: ViteNodeRunner) {
  const promisedReporters = reporterReferences.map(async (referenceOrInstance) => {
    if (typeof referenceOrInstance === 'string') {
      if (referenceOrInstance === 'html') {
        await ensurePackageInstalled('@vitest/ui', runner.root)
        const CustomReporter = await loadCustomReporterModule('@vitest/ui/reporter', runner)
        return new CustomReporter()
      }
      else if (referenceOrInstance in ReportersMap) {
        const BuiltinReporter = ReportersMap[referenceOrInstance as BuiltinReporters]
        return new BuiltinReporter()
      }
      else {
        const CustomReporter = await loadCustomReporterModule(referenceOrInstance, runner)
        return new CustomReporter()
      }
    }
    return referenceOrInstance
  })
  return Promise.all(promisedReporters)
}

function createBenchmarkReporters(reporterReferences: Array<string | Reporter | BenchmarkBuiltinReporters>, runner: ViteNodeRunner) {
  const promisedReporters = reporterReferences.map(async (referenceOrInstance) => {
    if (typeof referenceOrInstance === 'string') {
      if (referenceOrInstance in BenchmarkReportsMap) {
        const BuiltinReporter = BenchmarkReportsMap[referenceOrInstance as BenchmarkBuiltinReporters]
        return new BuiltinReporter()
      }
      else {
        const CustomReporter = await loadCustomReporterModule(referenceOrInstance, runner)
        return new CustomReporter()
      }
    }
    return referenceOrInstance
  })
  return Promise.all(promisedReporters)
}

export { createReporters, createBenchmarkReporters }
