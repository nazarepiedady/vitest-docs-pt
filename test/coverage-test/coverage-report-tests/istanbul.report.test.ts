/*
 * Istanbul coverage provider specific test cases
 */

import { expect, test } from 'vitest'
import { readCoverageJson } from './utils'

test('istanbul json report', async () => {
  const jsonReport = await readCoverageJson()

  // If this fails, you can use "npx live-server@1.2.1 ./coverage" to see coverage report
  expect(jsonReport).toMatchSnapshot()
})

test('implicit else is included in branch count', async () => {
  const coverageMap = await readCoverageJson()

  const filename = '<process-cwd>/src/implicitElse.ts'
  const fileCoverage = coverageMap[filename]

  expect(fileCoverage.b).toHaveProperty('0')
  expect(fileCoverage.b['0']).toHaveLength(2)
})

test('ignored code is excluded from the report', async () => {
  const functionName = 'ignoredFunction'
  const filename = '<process-cwd>/src/utils.ts'

  const coverageMap = await readCoverageJson()
  const fileCoverage = coverageMap[filename]

  // Function should not be included in report
  const functionCoverage = Object.values(fileCoverage.fnMap).find(fn => fn.name === functionName)
  expect(functionCoverage).toBe(undefined)

  // Function should still be found from the actual sources
  const utils = await import('../src/utils')
  expect(utils[functionName]).toBeTypeOf('function')
})
