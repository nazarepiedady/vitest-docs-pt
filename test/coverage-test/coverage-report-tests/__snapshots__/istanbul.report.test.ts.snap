// Vitest Snapshot v1

exports[`istanbul json report 1`] = `
{
  "<process-cwd>/src/Counter/Counter.component.ts": {
    "b": {},
    "branchMap": {},
    "f": {
      "0": 1,
      "1": 0,
      "2": 2,
      "3": 0,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 10,
            "line": 6,
          },
          "start": {
            "column": 2,
            "line": 6,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 9,
          },
          "start": {
            "column": 10,
            "line": 6,
          },
        },
        "name": "(anonymous_0)",
      },
      "1": {
        "decl": {
          "end": {
            "column": 22,
            "line": 12,
          },
          "start": {
            "column": 4,
            "line": 12,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 14,
          },
          "start": {
            "column": 22,
            "line": 12,
          },
        },
        "name": "(anonymous_1)",
      },
      "2": {
        "decl": {
          "end": {
            "column": 20,
            "line": 16,
          },
          "start": {
            "column": 4,
            "line": 16,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 18,
          },
          "start": {
            "column": 20,
            "line": 16,
          },
        },
        "name": "(anonymous_2)",
      },
      "3": {
        "decl": {
          "end": {
            "column": 37,
            "line": 20,
          },
          "start": {
            "column": 4,
            "line": 20,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 22,
          },
          "start": {
            "column": 37,
            "line": 20,
          },
        },
        "name": "(anonymous_3)",
      },
    },
    "path": "<process-cwd>/src/Counter/Counter.component.ts",
    "s": {
      "0": 1,
      "1": 1,
      "2": 0,
      "3": 2,
      "4": 0,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 7,
        },
        "start": {
          "column": 18,
          "line": 7,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 8,
        },
        "start": {
          "column": 4,
          "line": 8,
        },
      },
      "2": {
        "end": {
          "column": null,
          "line": 13,
        },
        "start": {
          "column": 6,
          "line": 13,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 17,
        },
        "start": {
          "column": 6,
          "line": 17,
        },
      },
      "4": {
        "end": {
          "column": null,
          "line": 21,
        },
        "start": {
          "column": 6,
          "line": 21,
        },
      },
    },
  },
  "<process-cwd>/src/Counter/Counter.vue": {
    "b": {
      "0": [
        2,
        1,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 18,
            "line": 8,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 8,
            },
            "start": {
              "column": 18,
              "line": 8,
            },
          },
          {
            "end": {
              "column": null,
              "line": 8,
            },
            "start": {
              "column": 18,
              "line": 8,
            },
          },
        ],
        "type": "binary-expr",
      },
    },
    "f": {
      "0": 1,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 18,
            "line": 8,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 18,
            "line": 8,
          },
        },
        "name": "(anonymous_1)",
      },
    },
    "path": "<process-cwd>/src/Counter/Counter.vue",
    "s": {
      "0": 1,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 8,
        },
        "start": {
          "column": 18,
          "line": 8,
        },
      },
    },
  },
  "<process-cwd>/src/Defined.vue": {
    "b": {
      "0": [
        1,
        0,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 0,
            "line": 5,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 8,
            },
            "start": {
              "column": 0,
              "line": 5,
            },
          },
          {
            "end": {
              "column": null,
              "line": 8,
            },
            "start": {
              "column": 2,
              "line": 8,
            },
          },
        ],
        "type": "if",
      },
    },
    "f": {},
    "fnMap": {},
    "path": "<process-cwd>/src/Defined.vue",
    "s": {
      "0": 1,
      "1": 1,
      "2": 1,
      "3": 0,
      "4": 1,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 2,
        },
        "start": {
          "column": 16,
          "line": 2,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 8,
        },
        "start": {
          "column": 0,
          "line": 5,
        },
      },
      "2": {
        "end": {
          "column": null,
          "line": 6,
        },
        "start": {
          "column": 2,
          "line": 6,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 8,
        },
        "start": {
          "column": 2,
          "line": 8,
        },
      },
      "4": {
        "end": {
          "column": null,
          "line": 12,
        },
        "start": {
          "column": 5,
          "line": 12,
        },
      },
    },
  },
  "<process-cwd>/src/Hello.vue": {
    "b": {
      "0": [
        3,
        1,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": 30,
            "line": 14,
          },
          "start": {
            "column": 16,
            "line": 14,
          },
        },
        "locations": [
          {
            "end": {
              "column": 18,
              "line": 14,
            },
            "start": {
              "column": 16,
              "line": 14,
            },
          },
          {
            "end": {
              "column": 30,
              "line": 14,
            },
            "start": {
              "column": 16,
              "line": 14,
            },
          },
        ],
        "type": "binary-expr",
      },
    },
    "f": {
      "0": 3,
      "1": 2,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 30,
            "line": 7,
          },
          "start": {
            "column": 24,
            "line": 7,
          },
        },
        "loc": {
          "end": {
            "column": 55,
            "line": 7,
          },
          "start": {
            "column": 30,
            "line": 7,
          },
        },
        "name": "(anonymous_1)",
      },
      "1": {
        "decl": {
          "end": {
            "column": 18,
            "line": 14,
          },
          "start": {
            "column": 16,
            "line": 14,
          },
        },
        "loc": {
          "end": {
            "column": 30,
            "line": 14,
          },
          "start": {
            "column": 18,
            "line": 14,
          },
        },
        "name": "(anonymous_3)",
      },
    },
    "path": "<process-cwd>/src/Hello.vue",
    "s": {
      "0": 1,
      "1": 1,
      "2": 3,
      "3": 1,
      "4": 2,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 6,
        },
        "start": {
          "column": 14,
          "line": 6,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 7,
        },
        "start": {
          "column": 15,
          "line": 7,
        },
      },
      "2": {
        "end": {
          "column": 55,
          "line": 7,
        },
        "start": {
          "column": 30,
          "line": 7,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 9,
        },
        "start": {
          "column": 0,
          "line": 9,
        },
      },
      "4": {
        "end": {
          "column": 30,
          "line": 14,
        },
        "start": {
          "column": 18,
          "line": 14,
        },
      },
    },
  },
  "<process-cwd>/src/implicitElse.ts": {
    "b": {
      "0": [
        1,
        0,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": null,
            "line": 5,
          },
          "start": {
            "column": 2,
            "line": 4,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 5,
            },
            "start": {
              "column": 2,
              "line": 4,
            },
          },
          {
            "end": {
              "column": null,
              "line": 5,
            },
            "start": {
              "column": 2,
              "line": 4,
            },
          },
        ],
        "type": "if",
      },
    },
    "f": {
      "0": 1,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 29,
            "line": 1,
          },
          "start": {
            "column": 16,
            "line": 1,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 49,
            "line": 1,
          },
        },
        "name": "implicitElse",
      },
    },
    "path": "<process-cwd>/src/implicitElse.ts",
    "s": {
      "0": 1,
      "1": 1,
      "2": 1,
      "3": 1,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 2,
        },
        "start": {
          "column": 10,
          "line": 2,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 5,
        },
        "start": {
          "column": 2,
          "line": 4,
        },
      },
      "2": {
        "end": {
          "column": null,
          "line": 5,
        },
        "start": {
          "column": 4,
          "line": 5,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 7,
        },
        "start": {
          "column": 2,
          "line": 7,
        },
      },
    },
  },
  "<process-cwd>/src/importEnv.ts": {
    "b": {},
    "branchMap": {},
    "f": {
      "0": 1,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 31,
            "line": 1,
          },
          "start": {
            "column": 16,
            "line": 1,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 3,
          },
          "start": {
            "column": 31,
            "line": 1,
          },
        },
        "name": "useImportEnv",
      },
    },
    "path": "<process-cwd>/src/importEnv.ts",
    "s": {
      "0": 1,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 2,
        },
        "start": {
          "column": 2,
          "line": 2,
        },
      },
    },
  },
  "<process-cwd>/src/index.mts": {
    "b": {},
    "branchMap": {},
    "f": {
      "0": 1,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 27,
            "line": 5,
          },
          "start": {
            "column": 16,
            "line": 5,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 7,
          },
          "start": {
            "column": 49,
            "line": 5,
          },
        },
        "name": "pythagoras",
      },
    },
    "path": "<process-cwd>/src/index.mts",
    "s": {
      "0": 1,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 6,
        },
        "start": {
          "column": 2,
          "line": 6,
        },
      },
    },
  },
  "<process-cwd>/src/untested-file.ts": {
    "b": {
      "0": [
        0,
        0,
      ],
      "1": [
        0,
        0,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": null,
            "line": 24,
          },
          "start": {
            "column": 2,
            "line": 21,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 24,
            },
            "start": {
              "column": 2,
              "line": 21,
            },
          },
          {
            "end": {
              "column": null,
              "line": 24,
            },
            "start": {
              "column": 2,
              "line": 21,
            },
          },
        ],
        "type": "if",
      },
      "1": {
        "loc": {
          "end": {
            "column": null,
            "line": 29,
          },
          "start": {
            "column": 2,
            "line": 26,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 29,
            },
            "start": {
              "column": 2,
              "line": 26,
            },
          },
          {
            "end": {
              "column": null,
              "line": 29,
            },
            "start": {
              "column": 2,
              "line": 26,
            },
          },
        ],
        "type": "if",
      },
    },
    "f": {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 39,
            "line": 6,
          },
          "start": {
            "column": 24,
            "line": 6,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 8,
          },
          "start": {
            "column": 39,
            "line": 6,
          },
        },
        "name": "untestedFile",
      },
      "1": {
        "decl": {
          "end": {
            "column": 13,
            "line": 10,
          },
          "start": {
            "column": 9,
            "line": 10,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 13,
          },
          "start": {
            "column": 35,
            "line": 10,
          },
        },
        "name": "add",
      },
      "2": {
        "decl": {
          "end": {
            "column": 18,
            "line": 15,
          },
          "start": {
            "column": 9,
            "line": 15,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 18,
          },
          "start": {
            "column": 40,
            "line": 15,
          },
        },
        "name": "multiply",
      },
      "3": {
        "decl": {
          "end": {
            "column": 21,
            "line": 20,
          },
          "start": {
            "column": 16,
            "line": 20,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 33,
          },
          "start": {
            "column": 64,
            "line": 20,
          },
        },
        "name": "math",
      },
    },
    "path": "<process-cwd>/src/untested-file.ts",
    "s": {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0,
      "6": 0,
      "7": 0,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 7,
        },
        "start": {
          "column": 2,
          "line": 7,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 12,
        },
        "start": {
          "column": 2,
          "line": 12,
        },
      },
      "2": {
        "end": {
          "column": null,
          "line": 17,
        },
        "start": {
          "column": 2,
          "line": 17,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 24,
        },
        "start": {
          "column": 2,
          "line": 21,
        },
      },
      "4": {
        "end": {
          "column": null,
          "line": 23,
        },
        "start": {
          "column": 4,
          "line": 23,
        },
      },
      "5": {
        "end": {
          "column": null,
          "line": 29,
        },
        "start": {
          "column": 2,
          "line": 26,
        },
      },
      "6": {
        "end": {
          "column": null,
          "line": 28,
        },
        "start": {
          "column": 4,
          "line": 28,
        },
      },
      "7": {
        "end": {
          "column": null,
          "line": 32,
        },
        "start": {
          "column": 2,
          "line": 32,
        },
      },
    },
  },
  "<process-cwd>/src/utils.ts": {
    "b": {
      "0": [
        0,
        1,
      ],
    },
    "branchMap": {
      "0": {
        "loc": {
          "end": {
            "column": null,
            "line": 16,
          },
          "start": {
            "column": 2,
            "line": 15,
          },
        },
        "locations": [
          {
            "end": {
              "column": null,
              "line": 16,
            },
            "start": {
              "column": 2,
              "line": 15,
            },
          },
          {
            "end": {
              "column": null,
              "line": 16,
            },
            "start": {
              "column": 2,
              "line": 15,
            },
          },
        ],
        "type": "if",
      },
    },
    "f": {
      "0": 1,
      "1": 2,
      "2": 0,
      "3": 1,
      "4": 0,
    },
    "fnMap": {
      "0": {
        "decl": {
          "end": {
            "column": 20,
            "line": 1,
          },
          "start": {
            "column": 16,
            "line": 1,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 3,
          },
          "start": {
            "column": 42,
            "line": 1,
          },
        },
        "name": "add",
      },
      "1": {
        "decl": {
          "end": {
            "column": 25,
            "line": 5,
          },
          "start": {
            "column": 16,
            "line": 5,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 7,
          },
          "start": {
            "column": 47,
            "line": 5,
          },
        },
        "name": "multiply",
      },
      "2": {
        "decl": {
          "end": {
            "column": 23,
            "line": 9,
          },
          "start": {
            "column": 16,
            "line": 9,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 12,
          },
          "start": {
            "column": 45,
            "line": 9,
          },
        },
        "name": "divide",
      },
      "3": {
        "decl": {
          "end": {
            "column": 21,
            "line": 14,
          },
          "start": {
            "column": 16,
            "line": 14,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 19,
          },
          "start": {
            "column": 32,
            "line": 14,
          },
        },
        "name": "sqrt",
      },
      "4": {
        "decl": {
          "end": {
            "column": 22,
            "line": 21,
          },
          "start": {
            "column": 16,
            "line": 21,
          },
        },
        "loc": {
          "end": {
            "column": null,
            "line": 24,
          },
          "start": {
            "column": 22,
            "line": 21,
          },
        },
        "name": "run",
      },
    },
    "path": "<process-cwd>/src/utils.ts",
    "s": {
      "0": 1,
      "1": 2,
      "2": 0,
      "3": 1,
      "4": 0,
      "5": 1,
      "6": 0,
    },
    "statementMap": {
      "0": {
        "end": {
          "column": null,
          "line": 2,
        },
        "start": {
          "column": 2,
          "line": 2,
        },
      },
      "1": {
        "end": {
          "column": null,
          "line": 6,
        },
        "start": {
          "column": 2,
          "line": 6,
        },
      },
      "2": {
        "end": {
          "column": null,
          "line": 11,
        },
        "start": {
          "column": 2,
          "line": 11,
        },
      },
      "3": {
        "end": {
          "column": null,
          "line": 16,
        },
        "start": {
          "column": 2,
          "line": 15,
        },
      },
      "4": {
        "end": {
          "column": null,
          "line": 16,
        },
        "start": {
          "column": 4,
          "line": 16,
        },
      },
      "5": {
        "end": {
          "column": null,
          "line": 18,
        },
        "start": {
          "column": 2,
          "line": 18,
        },
      },
      "6": {
        "end": {
          "column": null,
          "line": 23,
        },
        "start": {
          "column": 2,
          "line": 23,
        },
      },
    },
  },
}
`;
